index.js

```diff
@@ -615,8 +615,15 @@ async function getRepos(page = 1) {
  * @returns The manifest object
  */
 async function getRepoManifest(user, repo, branch) {
-    const url = `https://raw.githubusercontent.com/${user}/${repo}/${branch}/manifest.json`;
+    if (window.sessionStorage.getItem(user + "-" + repo)) {
+        return JSON.parse(window.sessionStorage.getItem(user + "-" + repo));
```

Comment:
This makes it only grab from sessionStorage once. 
```suggestion
    const sessionStorageItem = window.sessionStorage.getItem(`${user}-${repo}`);
    if (sessionStorageItem) {
        return JSON.parse(sessionStorageItem);
```